apps/member-profile/app/routes/_profile.profile.add-integration.ts

```diff
@@ -0,0 +1,54 @@
+import { json, type LoaderFunctionArgs, redirect } from '@remix-run/node';
+import { ensureUserAuthenticated, user } from 'app/shared/session.server';
+import { toast } from 'app/shared/session.server';
+
+import { addGithub } from '@/member-profile.server';
+import { GITHUB_API, GITHUB_TOKEN_URL, Route } from '../shared/constants';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const session = await ensureUserAuthenticated(request);
+  const id = user(session);
+
+  const url = new URL(request.url);
+  const code = url.searchParams.get('code');
+  // to be used for differentiating which service we are getting calledback from
+  const service = url.searchParams.get('service');
+
+  if (code) {
+    try {
+      const bodyData = new FormData();
+
+      bodyData.append('code', code);
+      bodyData.append('client_id', process.env.GITHUB_OAUTH_CLIENT_ID);
```

Comment:
I think these two environment variables need to first be defined in [env.ts](https://github.com/colorstackorg/oyster/blob/main/apps/api/src/shared/env.ts#L1), so that way we can use it in this file. We'll then want to add the following two lines in this file 
```typescript
 const GITHUB_OAUTH_CLIENT_ID = process.env.GITHUB_OAUTH_CLIENT_ID as string
 const GITHUB_OAUTH_SECRET = process.env.GITHUB_OAUTH_SECRET as string
```
This will make sure that the types are properly inferred, and should resolve the current issues with `Type | Undefined`.